src/orchard/complete.clj

```diff
@@ -0,0 +1,25 @@
+(ns orchard.complete
+  (:require [cljs-tooling.complete :as cljs-complete]
+            [compliment.core :as jvm-complete]
+            [compliment.utils :as jvm-complete-utils]
+            [orchard.misc :as u]
+            [orchard.util.cljs :as cljs]))
+
+(defn complete
+  [{:keys [ns symbol context extra-metadata] :as msg}]
+  (let [ns (u/as-sym ns)
+        prefix (str symbol)
+        extra-metadata (set (map keyword extra-metadata))]
+    (if-let [cljs-env (cljs/grab-cljs-env msg)]
```

Comment:
I guess to make this portable outside nREPL we should use `cljc`to check whether we're dealing with a Java or JavaScript completion.

Reply:
On the other hand - I'm assuming that with piggieback we'd actually be in `clj`, so probably this has to be some optional param we could pass (`cljs-env`).